### PR TITLE
[tools/depends] native and configure fixes

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Fixes some old variable usage (zlib)
Introduce usage of CMAKE_CXX_COMPILER_LAUNCHER for flatbuffer usage of ccache
assorted other fixes/changes to support Apple Silicon build host

## Motivation and context
Several fixes/cleanups to enable Apple silicon build host.

## How has this been tested?
Successful native dependency build on Apple Silicon host

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by garbear — APPROVED** [review]
ccache support for flatbuffers is a welcome change.

**Comment by fuzzard** [comment]
After a bit of work, have green across the board.
With this changeset, native depends builds on Apple silicon, and I can cross compile an x86_64 build that runs in Rosetta.

No more commits for this patch set, I'll wait for this to get approval and then go back to working through apple silicon target build

File: tools/depends/configure.ac
Comment (kambala-decapitator) [inline tools/depends/configure.ac:186]: should be either `-f` or `--find` per `xcrun`'s manual, there's no `-find` (single dash version)

Comment (fuzzard) [reply tools/depends/configure.ac:186]: Odd, it seems to work on Monterey. I'll go back and have a look at it

Comment (fuzzard) [reply tools/depends/configure.ac]: single dash operators all worked. Ive changed all to match double dash though just for uniformity of documented man

File: cmake/modules/FindCCache.cmake
Comment (fuzzard) [inline cmake/modules/FindCCache.cmake:17]: @wsnipex Is this correct, in particular the caching?

I found without the CACHE STRING "" FORCE that the launcher items werent listed in cmakecache.txt. 
Because apple uses wrapper scripts, i havent been able to test which is appropriate

Comment (wsnipex) [reply cmake/modules/FindCCache.cmake:17]: yes, that looks fine, we'd want that in CMakeCache.txt

**Pull request merged**
